File: packages/dart/lib/src/name/name_supports_middle_name.dart
import 'package:randino/src/name/data/index.dart';
import 'package:randino/src/types.dart';

/// Whether the language uses a middle name.
///
/// `includeMiddleName` is ignored for languages that do not — Korean, Japanese
/// and Chinese names have no middle part. A null [language] answers for the
/// mixed draw, where some of the languages do.
///
/// ```dart
/// nameSupportsMiddleName(NameLanguage.en); // true
/// nameSupportsMiddleName(NameLanguage.ko); // false
/// ```
bool nameSupportsMiddleName([NameLanguage? language]) =>
    language == null || nameData[language]!.hasMiddle;

File: packages/javascript/lib/name/nameSupportsMiddleName.ts
import type { NameLanguageOption } from '../_types/global.js';
import { NAME_DATA, resolveNameLanguage } from './data/index.js';

/**
 * Whether the language uses a middle name. `includeMiddleName` is ignored for
 * languages that do not — Korean, Japanese and Chinese names have no middle part.
 *
 * @example
 * nameSupportsMiddleName('en'); // true
 * nameSupportsMiddleName('ko'); // false
 */
export function nameSupportsMiddleName(language: NameLanguageOption = 'all'): boolean {
	const wanted = resolveNameLanguage(language);

	return wanted === 'all' ? true : NAME_DATA[wanted].hasMiddle;
}
